nba_core/app.py

```diff
@@ -0,0 +1,68 @@
+import os
+
+from flask import Flask
+
+from dal.database import connect_db
+from endpoints.endpoint_functions import before_request_endpoint, after_request_endpoint, user_signup_endpoint, home_endpoint, login_endpoint, logout_endpoint, get_player_stats_endpoint, get_adv_player_stats_endpoint
+
+app = Flask(__name__)
+
+app.config['SQLALCHEMY_DATABASE_URI'] = os.environ.get(
+    'DATABASE_URL', 'postgresql:///nba-users')
+app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
+app.config['SQLALCHEMY_ECHO'] = True
+app.config['SECRET_KEY'] = os.environ.get('SECRET_KEY', 'oneonetwotwo')
+
+connect_db(app)
+
+
+@app.before_request
+def add_user_to_g():
+    """If user is logged in, add curr user to Flask global."""
+    before_request_endpoint()
+
+
+@ app.after_request
+def add_header(req):
+    """Add non-caching headers on every request."""
+    return after_request_endpoint(req)
+
+######################################################################
+# General routes
+
+
+@app.route('/')
+def home():
+    # get the season leaders' stats that should be stored in database
+    return home_endpoint()
+
+
+@app.route('/signup', methods=["GET", "POST"])
+def user_signup():
+    """Handle user signup."""
+    return user_signup_endpoint()
+
+
+@app.route('/login', methods=["GET", "POST"])
+def login():
+    """Handle login of user."""
+    return login_endpoint()
+
+
+@app.route('/logout')
+def logout():
+    """Handle logout of user."""
+    return logout_endpoint()
+
+
+######################################################################
+# player search routes
+
+@app.route('/api/get-player-stats', methods=["GET", "POST"])
```

Comment:
How you name your routes is weirdly important to people. Its another thing where there are standard patterns that are accepted. One of these standards is that you don't put `get|put|post|delete` in your route -- it should be implied and understood by the http method. Consider this:

```
GET /api/get-player-stats
POST /api/add-player-stats
PUT /api/update-player-stats/{id}
PATCH /api/patch-player-stats/{id}
DELETE /api/delete-player-stats/{id}
```
vs.
```
GET /api/player-stats
POST /api/player-stats
PUT /api/player-stats/{id}
PATCH /api/player-stats/{id}
DELETE /api/player-stats/{id}
```

The later is the standard convention.